integration/network/network.go

```diff
@@ -0,0 +1,173 @@
+package network
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+	"net"
+	"os"
+	"strconv"
+
+	"github.com/giantswarm/ipam"
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	EnvCircleBuildNumber = "CIRCLE_BUILD_NUM"
+	EnvE2ECIDR           = "E2E_CIDR"
+	EnvE2EMask           = "E2E_MASK"
+
+	EnvAzureCIDR             = "AZURE_CIDR"
+	EnvAzureCalicoSubnetCIDR = "AZURE_CALICO_SUBNET_CIDR"
+	EnvAzureMasterSubnetCIDR = "AZURE_MASTER_SUBNET_CIDR"
+	EnvAzureWorkerSubnetCIDR = "AZURE_WORKER_SUBNET_CIDR"
+
+	AzureCalicoSubnetMask = 17
+	AzureMasterSubnetMask = 24
+	AzureWorkerSubnetMask = 24
+)
+
+func init() {
+	var err error
+
+	var azureNetwork *net.IPNet
+	{
+		azureNetwork, err = AzureNetwork()
+		if err != nil {
+			panic(err)
+		}
+		os.Setenv(EnvAzureCIDR, azureNetwork.String())
+	}
+
+	var azureMasterSubnet *net.IPNet
+	{
+		azureMasterSubnet, err = AzureSubnet(EnvAzureMasterSubnetCIDR, *azureNetwork, AzureMasterSubnetMask)
+		if err != nil {
+			panic(err)
+		}
+		os.Setenv(EnvAzureMasterSubnetCIDR, azureMasterSubnet.String())
+	}
+
+	var azureWorkerSubnet *net.IPNet
+	{
+		azureWorkerSubnet, err = AzureSubnet(EnvAzureWorkerSubnetCIDR, *azureNetwork, AzureWorkerSubnetMask, *azureMasterSubnet)
+		if err != nil {
+			panic(err)
+		}
+		os.Setenv(EnvAzureWorkerSubnetCIDR, azureWorkerSubnet.String())
+	}
+
+	var azureCalicoSubnet *net.IPNet
+	{
+		azureCalicoSubnet, err = AzureSubnet(EnvAzureCalicoSubnetCIDR, *azureNetwork, AzureCalicoSubnetMask, *azureMasterSubnet, *azureWorkerSubnet)
+		if err != nil {
+			panic(err)
+		}
+		os.Setenv(EnvAzureCalicoSubnetCIDR, azureCalicoSubnet.String())
+	}
+}
+
+// AzureNetwork either return network from CIDR found at EnvAzureCIDR environement variable or
+// determine network using EnvE2ECIDR, EnvE2EMask and EnvCircleBuildNumber.
+func AzureNetwork() (azureNetwork *net.IPNet, err error) {
+	azureCDIR := os.Getenv(EnvAzureCIDR)
+	if azureCDIR == "" {
+		e2eCIDR := os.Getenv(EnvE2ECIDR)
```

Comment:
Why is there an option here? Why not require one of those?

Reply:
`EnvE2ECIDR` is only required when `EnvAzureCIDR` is not provided. I can make this requirement explicit in init.

Reply:
The use case is the following: if the CIDR is provided we use it, otherwise we compute a value for it.

On CircleCI we will run multiple e2e tests at the same time and each test need to have its own CIDR to avoid network addresses collision.

Reply:
> if the CIDR is provided we use it

Why do we need to provide CIDR? Why can't we always use computed one?

> each test need to have its own CIDR to avoid network addresses collision

Why? Peering? What will be the created cluster peered with?